File: edTech-frontend/src/modules/exam/services/examService.ts
import axios from "axios";

const BASE_URL = "http://localhost:8080/api/exams";

export const getAllExams = () => {
    return axios.get(BASE_URL);
};

export const createExam = (exam: any) => {
    return axios.post(BASE_URL, exam);
};

export const getExamById = (id: string | number) => {
    return axios.get(`${BASE_URL}/${id}`);
};

export const updateExam = (id: string | number, exam: any) => {
    return axios.put(`${BASE_URL}/${id}`, exam);
};

export const deleteExam = (id: string | number) => {
    return axios.delete(`${BASE_URL}/${id}`);
};

File: backend/src/main/java/backend/com/edtech/exam/service/ExamService.java
package backend.com.edtech.exam.service;

import backend.com.edtech.exam.dto.ExamRequestDto;
import backend.com.edtech.exam.entity.Exam;

import java.util.List;

public interface ExamService {

    Exam createExam(ExamRequestDto dto);

    List<Exam> getAllExams();

    Exam getExamById(Long id);

    Exam updateExam(Long id, ExamRequestDto dto);

    void deleteExam(Long id);
}
